base/random.jl

```diff
@@ -1163,9 +1163,46 @@ randexp(rng::AbstractRNG, dims::Int...) = randexp!(rng, Array(Float64, dims))
 
 immutable UUID
     value::UInt128
+
+    UUID(u::UInt128) = new(u)
 end
-UUID(u::AbstractString) = convert(UUID, u)
 
+"""
+    uuid1([rng::AbstractRNG]) -> UUID
+
+Generates a version 1 (time-based) universally unique identifier (UUID), as specified
+by RFC 4122. Note that the Node ID is randomly generated (does not identify the host)
+according to section 4.5 of the RFC.
+"""
+function uuid1(rng::AbstractRNG=GLOBAL_RNG)
+    u = rand(rng, Uint128)
```

Comment:
this is using the deprecated capitalization of `UInt` types here and a few other places